crates/forge/src/main.rs

```diff
@@ -240,6 +241,34 @@ fn warn_if_snforge_std_not_compatible(scarb_metadata: &Metadata) -> Result<()> {
     Ok(())
 }
 
+fn maybe_save_tests_sierra_programs(
+    compiled_test_crates: &Vec<CompiledTestCrateRaw>,
+    execution_data_to_save: ExecutionDataToSave,
+    tests_programs_dir: &Utf8Path,
+    package_name: &str,
+) -> Result<HashMap<CrateLocation, TestProgramPath>> {
+    let save_test_sierra_program = match execution_data_to_save {
+        ExecutionDataToSave::Trace | ExecutionDataToSave::TraceAndProfile => true,
+        ExecutionDataToSave::None => false,
+    };
+    let mut map = HashMap::new();
+
+    for test_crate in compiled_test_crates {
+        let test_program_path = tests_programs_dir.join(format!(
+            "{package_name}_{:?}.sierra.json",
+            test_crate.tests_location
+        ));
+
+        if save_test_sierra_program {
+            let test_program_path =
+                TestProgramPath::save_test_program(&test_crate.sierra_program, test_program_path)?;
+            map.insert(test_crate.tests_location, test_program_path.clone());
+        }
+    }
```

Comment:
Please make it clearer what is the condition of saving the file. I would move `execution_data_to_save` as the first argument. Also I would yield `if save_test_sierra_program {` above the loop and create a separate, non-conditional saving function.

Reply:
Moved the `if` above, imo creating a non conditional function is not that beneficial